### PR TITLE
[mono][jit] Enabled more intrinsics for V4 on arm64

### PR DESCRIPTION
**Pull request by LeVladIonescu**
This PR contributes to https://github.com/dotnet/runtime/issues/86957.

- Passing element_type to emit_xequal instead of computing it from arg1->klass
- Get_Item, Set_item, Cctor will be addressed shortly after.

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -2815,7 +2792,11 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 	case SN_Negate:
 	case SN_op_UnaryNegation: {
 #if defined(TARGET_ARM64) || defined(TARGET_AMD64)
-		return emit_simd_ins (cfg, klass, OP_NEGATION, args [0]->dreg, -1);
+		if (!COMPILE_LLVM(cfg)) {
+			return emit_simd_ins_for_unary_op (cfg, klass, fsig, args, MONO_TYPE_R4, id);
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:2796]: Why is this condition needed?

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -2840,13 +2821,13 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 		if (!(fsig->param_count == 2 && mono_metadata_type_equal (fsig->params [0], type) && mono_metadata_type_equal (fsig->params [1], type)))
 			return NULL;
 		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
-		return emit_xequal (cfg, arg_class, args [0], args [1]);
+		return emit_xequal (cfg, arg_class, MONO_TYPE_R4, args [0], args [1]);
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:2824]: Why hard-coding the type?

Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:2824]: It is hard-coded for a lot of emit_vector_2_3_4 methods, those can only be of type Single.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:2796]: For !LLVM, arg_type is not being set if used `emit_simd_ins`, causing this assertion to be reached : https://github.com/dotnet/runtime/blob/15e9370e32ab7a2c25d56e95554ff69d334bff62/src/mono/mono/mini/mini-arm64.c#L467

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:2796]: Then, both of the LLVM and non-LLVM case could use `emit_simd_ins_for_unary_op`, right?

File: src/mono/mono/mini/simd-intrinsics.c (hunk 2 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:2824]: Let me re-frame my question. Is `get_underlying_type (m_class_get_this_arg (arg1->klass))` not returning `MONO_TYPE_R4` for `Vector4`?

Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c:2824]: Exactly, for `Vector4` that line translates to `get_underlying_type (MonoType* type)` which returns `type->type` , i.e `MONO_TYPE_VALUETYPE`.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:2824]: I see. Then I am okay with passing the type like this.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (LeVladIonescu) [reply src/mono/mono/mini/simd-intrinsics.c]: Went with the second option.

**Review by fanyang-mono — APPROVED** [review]
LGTM!

**Comment by LeVladIonescu** [comment]
Failures known

**Pull request merged**